scene/resources/particle_process_material.cpp

```diff
@@ -959,6 +959,15 @@ void ParticleProcessMaterial::_update_shader() {
 	code += "		// copied from previous version\n";
 	code += "		if (physics_params.damping > 0.0) {\n";
 	if (!particle_flags[PARTICLE_FLAG_DAMPING_AS_FRICTION]) {
+		code += "			float v = length(VELOCITY);\n";
+		code += "			v -= physics_params.damping * DELTA;\n";
+		code += "			if (v < 0.0) {\n";
+		code += "				VELOCITY = vec3(0.0);\n";
+		code += "			} else {\n";
+		code += "				VELOCITY = normalize(VELOCITY) * v;\n";
+		code += "			}\n";
```

Comment:
We should pull this out of the if/else conditional as it is shared by both approaches